src/Middleware/src/Headstart.API/Commands/EnvironmentSeed/EnvironmentSeedCommand.cs

```diff
@@ -42,39 +43,53 @@ public class EnvironmentSeedCommand : IEnvironmentSeedCommand
             IExchangeRatesCommand exhangeRates
         )
         {
-            _settings = settings;
             _portal = portal;
             _supplierCommand = supplierCommand;
             _buyerCommand = buyerCommand;
             _buyerLocationCommand = buyerLocationCommand;
             _oc = oc;
             _exhangeRates = exhangeRates;
-            var translationsConfig = new BlobServiceConfig()
-            {
-                ConnectionString = _settings.BlobSettings.ConnectionString,
-                Container = _settings.BlobSettings.ContainerNameTranslations
-            };
-            _translationsBlob = new OrderCloudIntegrationsBlobService(translationsConfig);
+            _settings = settings;
         }
 
         public async Task<EnvironmentSeedResponse> Seed(EnvironmentSeed seed)
         {
-            if (string.IsNullOrEmpty(_settings.OrderCloudSettings.ApiUrl))
+            if (string.IsNullOrEmpty(seed.OrderCloudSettings.Environment))
             {
-                throw new Exception("Missing required app setting OrderCloudSettings:ApiUrl");
+                throw new Exception("Missing required seeding field OrderCloudSettings:Environment");
             }
-            if (string.IsNullOrEmpty(_settings.OrderCloudSettings.WebhookHashKey))
+            if (string.IsNullOrEmpty(seed.OrderCloudSettings.WebhookHashKey))
             {
-                throw new Exception("Missing required app setting OrderCloudSettings:WebhookHashKey");
+                throw new Exception("Missing required seeding field OrderCloudSettings:WebhookHashKey");
             }
-            if (string.IsNullOrEmpty(_settings.EnvironmentSettings.MiddlewareBaseUrl))
+            var requestedEnv = validateEnvironment(seed.OrderCloudSettings.Environment);
+            if (requestedEnv == null)
```

Comment:
We can probably validate this and all other required properties on the model directly. For required properties just add the [Required] attribute.  Check out the `EnvironmentSeed` model  and see how I validated `InitialAdminPassword `  I think you can do something similar to validate ordercloud environment